### ISSUE TITLE
Embedded dashboard with optional locked filter doesn't work anymore when no value provided for this filter

### ISSUE THREAD

**Issue by ArnaudBab** [metabase/metabase#46375]
### Describe the bug

I have an embedded dashboard that has three filters (two are locked and one is editable).
We always provide a value for the first filter so no issue here. The third filter (a date filter) works fine and as expected.
The issue comes from the second filter which is an optional locked text filter. When in the Metabase dashboard, the dashboard loads fine without inputting any value in that filter. Though, when getting the url for the embedded dashboard from the API, when we do not provide a value for that filter, the API responds with a 400 and an issue that looks something like 'You must provide in the JWT a value for :commercantId' (the name of our filter).

This worked perfectly fine in v.0.50.13 but starting to fail as soon as I upgraded to v.0.50.18. As this is live in our production environment, I'd rather not try again until traffic on our website is lower so I don't have the exact logs. I had to downgrade back to v0.50.13 to make it work.

Thanks in advance for your help,
Arnaud Baboulin

### To Reproduce

1. Make sure you're in v0.50.18
2. Create a dashboard with an optional text filter
3. Publish it to make it available for embedding
4. Set that filter to locked 
5. Try getting the corresponding url from the API without giving any value to the filter

### Expected behavior

It should work as previously (v0.50.13), an optional filter should not prevent the dashboard to be loaded even if no value is provided for it

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "fr-FR",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.23+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.23",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.23+9",
    "os.name": "Linux",
    "os.version": "5.15.0-94-generic",
    "user.language": "en",
    "user.timezone": "UTC"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "14.12"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.3"
      }
    },
    "run-mode": "prod",
    "plan-alias": "",
    "version": {
      "date": "2024-07-15",
      "tag": "v0.50.13",
      "hash": "2086968"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

Blocking our users, preventing us from upgrading to any version after v0.50.13

**Comment by albertoperdomo** [comment]
Hello @ArnaudBab,

we recently tightened the usage of locked params to fix some other issues. The behavior you are describing (omitting the locked param value in the JWT) was unintended.

From [the docs](https://www.metabase.com/docs/latest/embedding/static-embedding-parameters#include-all-locked-parameters-in-your-server-code):

>Once you publish a chart or dashboard with a locked parameter, **you must include the name of the locked parameter in your server code.**

TL;DR
* If you include a locked filter, you need to pass a value in the JWT
* If you want to skip the effect of that locked filter, you can pass `[]` as value in the JWT

I think this is the change you need to do in order for it to work with the latest version. Please let me know if you have any follow up questions or comments.
